### PR TITLE
Jump Tags: add scripting hooks and examples

### PR DESCRIPTION
**Pull request by magicrub**
This PR extends PR https://github.com/ArduPilot/ardupilot/pull/22421 and adds two scripts:
- re-calibrate baro using rangefinder before you land
- check wind and jump to into-wind landing direction if better

### CODE REVIEW

**Comment by tridge** [comment]
we will wait till 1st one merges

File: libraries/AP_Scripting/docs/docs.lua [hunk 1]
```diff
@@ -1365,6 +1365,21 @@ function mission:state() end
...
+
+-- desc
+---@param tag integer
+---@return integer
+function mission:get_index_of_jump_tag(tag) end
+
+-- desc
+---@return integer|nil
+---@return integer|nil
+function follow:get_last_jump_tag() end
```
Comment (IamPete1) [inline libraries/AP_Scripting/docs/docs.lua:1381]:
```suggestion
function mission:get_last_jump_tag() end
```

Comment (magicrub) [reply libraries/AP_Scripting/docs/docs.lua]: fixed

**Comment by magicrub** [comment]
Lua changes from the merged PR https://github.com/ArduPilot/ardupilot/pull/22861 is causing LUA doc warnings which is now blocking this PR.. and any other PR that has LUA bindings changes

**Comment by peterbarker** [comment]
> Lua changes from the merged PR #22861 is causing LUA doc warnings which is now blocking this PR.. and any other PR that has LUA bindings changes

No, only the ones that try to have CI run again.  PRs that have already passed CI can be merged, breaking master ;-)

Not sure what the point of this comment was - there are new requirements, satisfy them?

File: libraries/AP_Scripting/docs/docs.lua [hunk 2]
```diff
@@ -1367,6 +1367,21 @@ function mission:state() end
...
+
+-- desc
+---@param tag integer
+---@return integer
+function mission:get_index_of_jump_tag(tag) end
+
+-- desc
+---@return integer|nil
+---@return integer|nil
+function mission:get_last_jump_tag() end
```
Comment (peterbarker) [inline libraries/AP_Scripting/docs/docs.lua:1383]: This documentation is wrong.

File: libraries/AP_Scripting/docs/docs.lua [hunk 3]
```diff
@@ -1493,6 +1508,10 @@ function baro:get_temperature() end
 ---@return number
 function baro:get_pressure() end
 
+-- desc
+---@return number
```
Comment (peterbarker) [inline libraries/AP_Scripting/docs/docs.lua:1512]: Kind of sparse documentation, right?  Knowing what *sort* of altitude is returned would be nice to have in the documentation (AMSL would be most outsider's guess, and *I'm* guess that that guess would be wrong).

Is there facility for more information in these documentation strings?

**Review by peterbarker — CHANGES_REQUESTED** [review]

File: libraries/AP_Scripting/docs/docs.lua (hunk 2 above)
Comment (magicrub) [reply libraries/AP_Scripting/docs/docs.lua:1383]: @IamPete1 please help

File: libraries/AP_Scripting/docs/docs.lua (hunk 3 above)
Comment (IamPete1) [reply libraries/AP_Scripting/docs/docs.lua:1512]: Yes, just add more text on the same line. There are some that are more thoroughly documented, would be good to get them all fully documented eventually.

File: libraries/AP_Scripting/docs/docs.lua (hunk 2 above)
Comment (IamPete1) [reply libraries/AP_Scripting/docs/docs.lua:1383]: If it passes CI you should be good. If it doesn't it should tell you what it expects.

Comment (magicrub) [reply libraries/AP_Scripting/docs/docs.lua:1383]: it passes CI so resolving

**Comment by magicrub** [comment]
@peterbarker @IamPete1 I've added all the comments from the libraries into the script desc for all of Baro and Mission

File: libraries/AP_Mission/AP_Mission.cpp [hunk 4]
```diff
@@ -2091,6 +2091,18 @@ uint16_t AP_Mission::get_index_of_jump_tag(const uint16_t tag) const
     return 0;
 }
 
+#if AP_SCRIPTING_ENABLED
+bool AP_Mission::get_last_jump_tag(uint16_t &tag, uint16_t &age) const
```
Comment (tridge) [inline libraries/AP_Mission/AP_Mission.cpp:2095]: needs a comment for units of age (the name makes it sound like a time, which is isn't)

File: libraries/AP_Scripting/docs/docs.lua [hunk 5]
```diff
@@ -1318,55 +1318,80 @@ function RPM:get_rpm(instance) end
 ---@field MISSION_STOPPED number
 mission = {}
 
--- desc
+-- clear - clears out mission
 ---@return boolean
 function mission:clear() end
 
--- desc
+-- set any WP items in any order in a mavlink-ish kinda way.
```
Comment (tridge) [inline libraries/AP_Scripting/docs/docs.lua:1325]: what does this comment mean? "mavlink-ish" ??

**Review by peterbarker — APPROVED** [review]
Looks good

**Review by tridge — APPROVED** [review]

**Pull request merged**